File: 0121.买卖股票的最佳时机/0121-买卖股票的最佳时机.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if prices==None or len(prices)==0:
            return 0
        result=0
        temp=0
        for i in range(1,len(prices)):
            temp+=prices[i]-prices[i-1]
            temp=max(temp,0)
            result=max(temp,result)
        return result

File: 0121.买卖股票的最佳时机/0121-买卖股票的最佳时机.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices==null || prices.length==0) return 0;
        int result=0;
        int min=prices[0];
        for (int i=1;i<prices.length;i++){
            result=(result>prices[i]-min)? result:prices[i]-min;
            min=(prices[i]<min)? prices[i]:min;
        }
        return result;
    }
}